nixos/modules/services/networking/pounce.nix

```diff
@@ -0,0 +1,224 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pounce;
+  defaultUser = "pounce";
+  configFile = pkgs.writeText "pounce" ''
+    ${optionalString (cfg.sslCa != null) "local-ca = "+cfg.sslCa}
+    local-host = ${cfg.hostAddress}
+    local-port = ${toString cfg.hostPort}
+    local-priv = ${cfg.sslKey}
+    local-cert = ${cfg.sslCert}
+    join = ${concatStringsSep "," cfg.joinChannels}
+    ${optionalString (cfg.hashedPass != "") "local-pass = "+cfg.hashedPass}
+    host = ${cfg.ircHost}
+    port = ${toString cfg.ircHostPort}
+    user = ${cfg.ircUser}
+    nick = ${cfg.ircNick}
+    ${optionalString (cfg.certFP != null) "client-cert = "+cfg.certFP+"\nsasl-external"}
+    ${optionalString (cfg.ircPass != "") "pass = "+cfg.ircPass}
+    ${cfg.extraConfig}
+  '';
+in
+{
+  options = {
+    services.pounce = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+        Whether to enable Pounce
+        '';
+      };
+
+      user = mkOption {
+        default = defaultUser;
+        example = "john";
+        type = types.str;
+        description = ''
+          The name of an existing user account to use to own the Pounce server
+          process. If not specified, a default user will be created.
+        '';
+      };
+
+      group = mkOption {
+        default = defaultUser;
+        example = "users";
+        type = types.str;
+        description = ''
+          Group to own the Pounce process.
+        '';
+      };
+
+      dataDir = mkOption {
+        type = types.path;
+        default = "/var/lib/pounce";
+        description = ''
+          The data directory for pounce.
+        '';
+      };
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.pounce;
+        defaultText = "pkgs.pounce";
+        description = ''
+          Set version of pounce package to use.
+        '';
+      };
+
+      hostAddress = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
```

Comment:
listening on IPv4 only by default is slowly becoming a bit anachronistic. maybe `localhost` instead, or does pounce on all addresses automatically when not told where to listen? (in that case `nullOr str` with `null` default would be fitting)